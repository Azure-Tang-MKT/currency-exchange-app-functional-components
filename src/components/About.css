.aboutText {
  padding-top: 30px;
  padding-left: 30px;
  padding-bottom: 30px;
  text-align: left;
}

button.backBtn {
  font-size: 18px;
  padding: 12px;
  border: 0ch;
  border-radius: 3px;
  float: left;
  margin-left: 30px;
}
