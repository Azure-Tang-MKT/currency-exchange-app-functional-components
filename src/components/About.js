import React from "react";
import "./About.css";
import { useNavigate } from "react-router-dom";

function About() {
  let navigate = useNavigate();
  function handleClick() {
    navigate(-1);
  }

  return (
    <div>
      <p className="aboutText">
        This is a web based app that helps user to know the exchange rate of
        some foreign currencies to Canadian dollar.
      </p>
      <button className="backBtn" onClick={handleClick}>
        Go Back
      </button>
    </div>
  );
}

export default About;
