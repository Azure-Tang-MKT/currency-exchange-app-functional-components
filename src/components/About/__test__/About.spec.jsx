import { render, screen } from "@testing-library/react";
import About from "../About";
import { MemoryRouter } from "react-router-dom";
import React from "react";

describe("About Page", () => {
  it("renders the GoBack page with page info and go back button", () => {
    const { container } = render(
      <MemoryRouter>
        <About />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    expect(
      screen.getByText(
        "This is a web based app that helps user to know the exchange rate of some foreign currencies to Canadian dollar."
      )
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Go Back" })).toBeInTheDocument();
  });
});
