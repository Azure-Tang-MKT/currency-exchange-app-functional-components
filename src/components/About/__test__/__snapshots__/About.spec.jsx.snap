// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page renders the GoBack page with page info and go back button 1`] = `
<div>
  <div>
    <p
      class="aboutText"
    >
      This is a web based app that helps user to know the exchange rate of some foreign currencies to Canadian dollar.
    </p>
    <button
      class="backBtn"
    >
      Go Back
    </button>
  </div>
</div>
`;
